synapse/storage/events.py

```diff
@@ -2033,7 +2033,7 @@ def is_event_after(self, event_id1, event_id2):
         to_2, so_2 = yield self._get_event_ordering(event_id2)
         defer.returnValue((to_1, so_1) > (to_2, so_2))
 
-    @defer.inlineCallbacks
+    @cachedInlineCallbacks(max_entries=5000)
```

Comment:
I'd really prefer it if we used
```
@cached()
@defer.inlineCallbacks
```

but I'm not going to argue too much about it

Reply:
Me too, but I think this is something to be investigated outside this PR.